src/tests/functional/plugin/shared/include/behavior/plugin/ov_auto_batching_tests.hpp

```diff
@@ -0,0 +1,165 @@
+            }
+            // minimize timeout to reduce test time
+            config[CONFIG_KEY(AUTO_BATCH_TIMEOUT)] = std::to_string(1);
+
+            auto compiled_model = core->compile_model(model, std::string(ov::test::utils::DEVICE_BATCH) + ":" +
+                                                      target_device + "(" + std::to_string(num_batch) + ")",
+                                                      config);
+
+            auto network_outputs = model->outputs();
+            ASSERT_EQ(network_outputs.size(), 1) << " Auto-Batching tests use networks with single output";
+            auto output = *network_outputs.begin();
+            for (size_t j = 0; j < num_requests; j++) {
+                outputs.push_back(output);
+                outElementsCount.push_back(
+                        std::accumulate(begin(fn_ptrs[i]->get_output_shape(0)), end(fn_ptrs[i]->get_output_shape(0)), 1,
+                                        std::multiplies<size_t>()));
+
+                auto inf_req = compiled_model.create_infer_request();
+                irs.push_back({model, inf_req});
+
+                auto compiled_model_ref = core->compile_model(model, ov::test::utils::DEVICE_TEMPLATE);
+                auto inf_req_ref = compiled_model_ref.create_infer_request();
+                irs_ref.push_back(inf_req_ref);
+
+                std::vector<ov::Tensor> inData;
+                for (auto input : inputs) {
+                    auto tensor = ov::test::utils::create_and_fill_tensor(input.get_element_type(), input.get_shape());
+                    if (use_get_tensor)
+                        memcpy(reinterpret_cast<void *>(inf_req.get_tensor(input).data()),
+                               reinterpret_cast<const void *>(tensor.data()), tensor.get_byte_size());
+                    else
+                        inf_req.set_tensor(input, tensor);
+
+                    inf_req_ref.set_tensor(input, tensor);
+                }
+
+                if (!use_get_tensor) {
+                    auto tensor = ov::test::utils::create_and_fill_tensor(output.get_element_type(), output.get_shape());
+                    inf_req.set_tensor(output, tensor);
+                }
+
+                inf_req_ref.infer();
+            }
+        }
+
+        const int niter = 1;
+        for (int i = 0; i < niter; i++) {
+            for (auto ir : irs) {
+                ir.second.start_async();
+            }
+
+            for (auto ir : irs) {
+                ir.second.wait();
+            }
+        }
+
+        for (size_t i = 0; i < irs.size(); ++i) {
+            auto output = irs[i].first->get_results().at(0);
+            auto out = irs[i].second.get_tensor(output);
+            auto out_ref = irs_ref[i].get_tensor(output);
+            ov::test::utils::compare(out_ref, out);
+        }
+    }
+};
+
+class AutoBatching_Test_DetectionOutput : public AutoBatching_Test {
+public:
+    void SetUp() override {
```

Comment:
Detail:
It will be better if `SetUp` is protected (is defined protected in `gtest` base class `Test`), but `getTestCaseName` must be public.

Reply:
fixed